### PR TITLE
Tournaments Page

### PR DESCRIPTION
**Pull request by lowtorola**
Adds the Tournaments and Tournament pages!

### CODE REVIEW

File: frontend2/src/views/Tournament.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        const result = await getTournamentMatches(
+          episodeId,
+          teamId ?? undefined,
+          tournamentId,
+          round,
+          page,
+        );
+        setMatches(result);
+      } catch (err) {
+        console.error(err);
```
Comment (lowtorola) [inline frontend2/src/views/Tournament.tsx:41]: `setData(undefined)`

File: frontend2/src/views/Tournament.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    };
+
+    void fetchMatches();
+  }, [teamId, page, round]);
+
+  return (
+    <div className="flex h-full w-full flex-col bg-white p-6">
+      {/* TODO: change to tournament name from tourney query and add LOADING COMPONENT!!! */}
+      <h1 className="text-2xl font-bold">Tournament {tournamentId}</h1>
+      <BattlecodeTable
```
Comment (lowtorola) [inline frontend2/src/views/Tournament.tsx:54]: Wrap on width-limiting `<div>`

File: frontend2/src/views/Tournaments.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import BattlecodeTable from "../components/BattlecodeTable";
+// import { CurrentUserContext } from "../contexts/CurrentUserContext";
+import Icon from "../components/elements/Icon";
+
+const isInternalLink = (to: string): boolean => {
+  const url = new URL(to, window.location.origin);
+  return url.hostname === window.location.hostname;
+};
+
+const MarkdownSection: React.FC<{ children: string; className?: string }> = ({
```
Comment (lowtorola) [inline frontend2/src/views/Tournaments.tsx:17]: Refactor to component called `Markdown.tsx` and make sure to accept className prop!

File: frontend2/src/views/Tournaments.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  useEffect(() => {
+    const fetchSchedule = async (): Promise<void> => {
+      if (!scheduleHasLoaded.current) {
+        scheduleHasLoaded.current = true;
+        setLoading(true);
+        try {
+          const result = await getAllTournaments(episodeId);
+          setSchedule(result.results);
+        } catch (err) {
+          console.error(err);
```
Comment (lowtorola) [inline frontend2/src/views/Tournaments.tsx:69]: `setData(undefined)`

File: frontend2/src/views/Tournaments.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+              </NavLink>
+            ),
+          },
+          {
+            header: "About",
+            value: (r) => <span>{r.blurb}</span>,
+          },
+        ]}
+      />
+      <MarkdownSection className="mt-10">
```
Comment (lowtorola) [inline frontend2/src/views/Tournaments.tsx:130]: Replace with a single `<Markdown>` component with these texts concatenated!

**Comment by lowtorola** [comment]
Low key I think this UI is my best yet... not good but maybe I'm less incompetent 🥲

**Review by acrantel — APPROVED** [review]
great work! lgtm, left some minor suggestions

**Pull request merged**